src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_CAPACITY = 16;
+    public static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        this.threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
```

Comment:
that's great that you divided these two operations. But could you please also create a separate method for index retrieval so that these two lines of code aren't duplicated in several methods?

Reply:
Changed, but not sure if it what you mean.